test-cypress/steps/graphql/edit-graphql-endpoint-steps.js

```diff
@@ -0,0 +1,100 @@
+export class EditGraphqlEndpointSteps {
+    static getModal() {
+        return cy.get('.graphql-endpoint-configuration-modal');
+    }
+
+    static getModalTitle() {
+        return this.getModal().find('.modal-title');
+    }
+
+    static getDynamicForm() {
+        return this.getModal().find('dynamic-form');
+    }
+
+    static getFormFields() {
+        return this.getDynamicForm().find('.form-field');
+    }
+
+    static getInputField(index) {
+        return this.getFormFields().find('.input-field input').eq(index);
+    }
+
+    static getBooleanField(index) {
+        return this.getFormFields().find('.boolean-field input').eq(index);
+    }
+
+    static getSelectField(index) {
+        return this.getFormFields().find('.select-field select').eq(index);
+    }
+
+    static getMultiSelectField(index) {
+        return this.getFormFields().find('.multiselect-field multiselect-dropdown').eq(index);
```

Comment:
All these steps where you take a filed by type and its index in the form can be quite fragile. We can just put ids on the filed based on the model and reference by those ids.